### PR TITLE
Add a rotating subset of members for early-cycle enkf

### PR DESCRIPTION
**Pull request by TravisElless-NOAA**
# Description

The early-cycle EnKF needs the ability to run with fewer members than the late-cycle due to operational resource constraints. Because of this requirement, the introduction of a rotating subset of member first-guess states used by the early-cycle ensemble is also needed in order to preserve the rotating member initial condition functionality currently used by the GEFS.
   
# Type of change

- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
-Cycled test on Orion

# Checklist
- [ ] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: scripts/exgdas_enkf_ecen.sh
Comment (aerorahul) [inline scripts/exgdas_enkf_ecen.sh]:
```suggestion
   if (( smem > 80 )); then
```

Comment (TravisElless-NOAA) [reply scripts/exgdas_enkf_ecen.sh]: Thanks for pointing me in the right direction

File: jobs/JGLOBAL_FORECAST
Comment (WalterKolczynski-NOAA) [inline jobs/JGLOBAL_FORECAST]:
```suggestion
export rCDUMP="${RUN/gfs/gdas}"
```

File: scripts/exgdas_enkf_post.sh
Comment (WalterKolczynski-NOAA) [inline scripts/exgdas_enkf_post.sh]:
```suggestion
if [[ "${RUN}" == "enkfgfs" ]]; then
```

File: scripts/exgdas_enkf_sfc.sh
Comment (WalterKolczynski-NOAA) [inline scripts/exgdas_enkf_sfc.sh]:
```suggestion
if [ "${RUN}" == "enkfgfs" ]; then
    NMEM_ENS=${NMEM_ENS_GFS:-30}
    ec_offset=${NMEM_ENS_GFS_OFFSET:-20}
    mem_offset=$((ec_offset * cyc/6))
else
    NMEM_ENS=${NMEM_ENS:-80}
    mem_offset=0
fi
```

File: scripts/exgdas_enkf_ecen.sh
Comment (WalterKolczynski-NOAA) [inline scripts/exgdas_enkf_ecen.sh:116]:
```suggestion
   smem=$(( (imem + mem_offset) % 80 ))
```

File: scripts/exgdas_enkf_update.sh
Comment (WalterKolczynski-NOAA) [inline scripts/exgdas_enkf_update.sh:189]:
```suggestion
   smem=$(( (imem + mem_offset) % 80 ))
```

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: scripts/exgdas_enkf_update.sh
Comment (TravisElless-NOAA) [reply scripts/exgdas_enkf_update.sh:189]: This change causes the scripts to fail. When `imem + mem_offset = 80` this causes the result of `smem=0` and causes the script to replace member 80 with member 00. Member 80 should still exist and then when `imem + mem_offset = 81` gets calculated it should go to member 1 instead of 0.

Comment (WalterKolczynski-NOAA) [reply scripts/exgdas_enkf_update.sh:189]: Yeah, you're right. Stupid 1-indexed arrays. Any way that keeps the modulus and works around it is just messier, so let's keep it how it was.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests. Please merge in this afternoon's develop updates and we will run it through.

**Comment by TravisElless-NOAA** [comment]
Thanks for your input on this @WalterKolczynski-NOAA

**Comment by WalterKolczynski-NOAA** [comment]
It occurred to me days after this was merged that it uses a hard-coded value for the late-cycle ensemble size, and will likely break for smaller ensemble sizes.

**Pull request merged**
